enhancements/baremetal-IPI/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,185 @@
+
+2. Naming the new CR.
+
+3. Since this enhancement did not make it into 4.3 and the BareMetal IPI
+deployments are in Tech Preview in 4.2, the upgrade strategy should include
+upgrades from 4.2 to 4.3 to 4.4. (Assuming this enhancement makes it into 4.4.)
+
+## Summary
+
+The configuration required to bring up a Baremetal IPI (metal3) cluster needs
+to be availabe in a Config Resource. We had submitted an earlier proposal [1]
+to augment the BareMetalInfrastructureStatus CR with these config items. Based
+on feedback we recieved we are revising our proposal to create a new CR for
+these configuration items.
+
+This new CR needs to be accessed by just the Installer and the Machine API
+Operator (API) and hence does not need to be created with global scope.
+
+## Motivation
+
+The Baremetal IPI deployments are different from the other platform types currently
+being supported by OpenShift in that there is no underlying cloud controller
+that exists until provisioning of baremetal servers takes place first.
+
+The "metal3" pod deployed by the machine API operator (MAO) contains the BaremMetal
+Operator (BMO) and a provisioning service (Ironic) that are together responsible for
+PXE booting baremetal servers and enrolling them as BareMetal Hosts.
+
+This enhancement request proposes to add a new CR for configs required only for
+the provisioning service to PXE boot a baremetal server and make it available as a
+BareMetal host.The configurations in this baremetal provisioning CR, are seperate
+from the configuration in the BareMetalHost CR. This new CR will allow for setting
+some sane defaults for these configurations with the ability to overrride them if 
+required. After the baremetal server is provisioned, the configurations in the
+BareMetalHost CR contain information to connect to the controller and also to
+fulfill a Machine.
+
+### Goals
+
+The goal of this enhancement request is to provide details about the configuration
+being added to a new CR used only in "metal3" deployments. This new CR would be
+within the operator scope and will only contain contain configuration required
+by the provisioning service to boot baremetal servers.
+
+### Non-Goals
+
+All config values except the RHCOSImageUrl are not expected to change after a
+metal3 cluster is up. Handling changes to this value after install time has
+not been explored and that is a non-goal at this time.
+Update lifecycle for bootimage is an unsolved problem for all platforms. For
+more information, refer to : https://github.com/openshift/os/issues/381.
+
+### Proposal
+
+Before BareMetal Hosts can be matched to Machines, they need to be connected to their
+provisioning network for them to be PXE booted and given an IP address. To make
+this happen, the provisioning service needs to know which NIC, provisioning network,
+IP address and image URLs to use to download and boot images on these servers.
+
+A new CR called "BareMetalController" or "Metal3Controller" would be created within
```

Comment:
We should use "Metal3" as the prefix for all public resources like this, so I would go with Metal3Controller.

Reply:
Makes sense.